### ISSUE TITLE
eosio-cpp -abigen on an action that has pass by reference argument results in type &

### ISSUE THREAD

**Issue by andriantolie** [EOSIO/eosio.cdt#81]
For example, the following action (taken from https://github.com/EOSIO/eosio-project-boilerplate-simple)
`void update( account_name _user, std::string& _note )`

Will result in the following abi
```
        {
            "name": "update",
            "base": "",
            "fields": [
                {
                    "name": "_user",
                    "type": "name"
                },
                {
                    "name": "_note",
                    "type": "&"
                }
            ]
        }
```

Notice that the above _note type is & instead of string

**Comment by sunqiLoveSnow** [comment]
same with me, well, when you change 'const string&' to 'string', the '&' in abi would be replaced with 'string'. But...still, invalide abi error exist when you set your contract..... anyone know how 2 solve it?

**Comment by larryk85** [comment]
should be resolved by #88
